src/dbus/methods/player_peer_info.c

```diff
@@ -34,38 +35,90 @@ void cmd_peer_player_info_dbus_cb(const char *result,
+        g_variant_builder_add(&builder, "{ss}", "items_unlocked", get_info(result, "items_unlocked='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "challenges_completed", get_info(result, "challenges_completed='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "missions_completed", get_info(result, "missions_completed='", "'", NULL));
+
+        g_variant_builder_add(&builder, "{ss}", "pvp_wins", get_info(result, "pvp_wins='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "pvp_loses", get_info(result, "pvp_loses='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "pvp_total_matches", get_info(result, "pvp_total_matches='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "pvp_kills", get_info(result, "pvp_kills='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "pvp_deaths", get_info(result, "pvp_deaths='", "'", NULL));
+
+        g_variant_builder_add(&builder, "{ss}", "playtime_seconds", get_info(result, "playtime_seconds='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "leavings_percentage", get_info(result, "leavings_percentage='", "'", NULL));
+
+        g_variant_builder_add(&builder, "{ss}", "coop_climbs_performed", get_info(result, "coop_climbs_performed='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "coop_assists_performed", get_info(result, "coop_assists_performed='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "favorite_pvp_class", get_info(result, "favorite_pvp_class='", "'", NULL));
+        g_variant_builder_add(&builder, "{ss}", "favorite_pve_class", get_info(result, "favorite_pve_class='", "'", NULL));
+
+        g_variant_builder_add(&builder, "{ss}", "clan_name", get_info(result, "clan_name='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "clan_role", get_info(result, "clan_role='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "clan_position", get_info(result, "clan_position='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "clan_points", get_info(result, "clan_points='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "clan_member_since", get_info(result, "clan_member_since='", "'", NULL) ? : "");
     }
 
+    playerinfodict = g_variant_builder_end(&builder);
+
     warfacebot_complete_player_peer_info(
         a->object,
         a->invocation,
-        result);
+        playerinfodict);
 
     g_free(a);
 }
 
-static void cmd_profile_info_get_status_dbus_cb(const char *info, void *args)
+void cmd_profile_info_get_status_dbus_cb(const char *info, void *args)
 {
     struct cb_args *a = (struct cb_args *) args;
 
     if (info == NULL)
     {
+        GVariantBuilder builder;
+        GVariant *playerinfodict;
+
+        g_variant_builder_init(&builder, G_VARIANT_TYPE("a{?*}"));
+        g_variant_builder_add(&builder, "{ss}", "error", "-2");
```

Comment:
I still don't get why one would need to differentiate (with different error values -1 / -2) errors coming from a player to an error of the server. In both cases it means the player is either offline or didn't git you an answer (same for the server). Maybe make the field a boolean? `error=true` for instance.